packages/bs-ethereum/src/services/blockchain-data/BlockscoutBDSEthereum.ts

```diff
@@ -415,4 +441,10 @@ export class BlockscoutBDSEthereum extends RpcBDSEthereum {
 
     return data.items[0].height
   }
+
+  #resolveFullTransactionsByAddressNetwork(): 'mainnet' | 'testnet' {
+    if (this._network.id === BSEthereumConstants.NEOX_TESTNET_NETWORK_ID) return 'testnet'
+
+    return 'mainnet'
+  }
```

Comment:
Do it inside the `getFullTransactionsByAddress` function. You are not using this and you are not doing a complex conditional.